doc/ExpertOptions.md

```diff
@@ -980,6 +1033,27 @@ configuration sets it to use native docker, and all applets are given
 the `allProjects: VIEW` permission. This allows them to access the
 credentials file, even if it is stored on a different project.
 
+### AWS ECR registries
+
+Logging into an AWS Elastic Container Registry (ECR) is a bit different
+than logging into a standard docker registry. Specifically, the AWS
+command line client is used to dynamically generate a password from an AWS 
+user profile. To handle this use-case, dxCompiler downloads the
+required AWS `credentials` file, installs the AWS client, and generates the 
+password. See the
+[AWS documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html)
+for more details and examples.
+
+```
+{
+  "dockerRegistry": {
+    "registry": "<aws_account_id>.dkr.ecr.<region>.amazonaws.com",
+    "credentials": "dx://myproj/aws_credentials",
+    "awsRegion": "us-east-1"
```

Comment:
Could you remove awsRegion from the spec and extract it from the registry suffix?

Reply:
I think it's a good idea to keep it. If region is specified, it also tells us that the registry is ECR. Otherwise we have to parse the registry name. If AWS ever changes the structure of their registry names, it breaks our code.

From Keith: "many AWS calls require a region to be specified, so having this defined and usable will reduce pain/refactoring as use-cases are added.  It's pretty standard to have this called out explicitly and rare to try to infer from another resource identifier.  +1 for leaving."